control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -173,8 +174,56 @@ func (r *HostedControlPlaneReconciler) SetupWithManager(mgr ctrl.Manager, create
 	return nil
 }
 
+func retrieveHostedCluster(ctx context.Context, c client.Client, hclusterNamespacedName string) *hyperv1.HostedCluster {
+	hcluster := &hyperv1.HostedCluster{}
+
+	err := c.Get(ctx, util.ParseNamespacedName(hclusterNamespacedName), hcluster)
+	if err != nil {
+		log := ctrl.LoggerFrom(ctx)
+		log.Info("Failed to retrieve cluster", "hclusterNamespacedName", hclusterNamespacedName, "err", err)
+
+		return nil
+	}
+
+	return hcluster
+}
+
 func (r *HostedControlPlaneReconciler) setup(createOrUpdate upsert.CreateOrUpdateFN) {
-	r.createOrUpdate = createOrUpdateWithOwnerRefFactory(createOrUpdate)
+	delegateFactory := createOrUpdateWithOwnerRefFactory(createOrUpdate)
+
+	r.createOrUpdate = func(owner *hyperv1.HostedControlPlane) upsert.CreateOrUpdateFN {
```

Comment:
nit: can you extract this in its own function like:
func createOrUpdateDelayScrapeConfig(createOrUpdate upsert.CreateOrUpdateFN) upsert.CreateOrUpdateFN {
... blah ...
}
and then call that function from setup ?

Reply:
Sure, I can do that. Thanks.